libs/wire-api/src/Wire/API/Routes/Named.hs

```diff
@@ -19,21 +19,40 @@
 
 module Wire.API.Routes.Named where
 
+import Control.Lens ((%~))
 import Data.Kind
 import Data.Metrics.Servant
 import Data.Proxy
+import Data.Swagger
 import GHC.TypeLits
 import Imports
 import Servant
 import Servant.Client
 import Servant.Client.Core (clientIn)
 import Servant.Swagger
 
-newtype Named named x = Named {unnamed :: x}
+newtype Named name x = Named {unnamed :: x}
   deriving (Functor)
 
-instance HasSwagger api => HasSwagger (Named name api) where
-  toSwagger _ = toSwagger (Proxy @api)
+-- | For 'HasSwagger' instance of 'Named'.  'KnownSymbol' isn't enough because we're using
+-- types other than string literals in some places.
+class RenderableSymbol a where
+  renderSymbol :: Text
+
+instance {-# OVERLAPPABLE #-} KnownSymbol a => RenderableSymbol a where
+  renderSymbol = cs $ symbolVal (Proxy @a)
+
+instance {-# OVERLAPPING #-} (RenderableSymbol a, RenderableSymbol b) => RenderableSymbol '(a, b) where
+  renderSymbol = "(" <> (renderSymbol @a) <> ", " <> (renderSymbol @b) <> ")"
+
+instance (HasSwagger api, RenderableSymbol name) => HasSwagger (Named name api) where
+  toSwagger _ =
+    toSwagger (Proxy @api)
+      & allOperations . summary %~ (<> Just dscr)
+      & allOperations . description %~ (Just (dscr <> "\n\n") <>)
```

Comment:
not sure we want both, but i can't decide which one to prefer?

Reply:
Let's keep the one in the description, not the summary